homework/hackyourtemperature/server.js

```
@@ -0,0 +1,41 @@
+import express from 'express';
+
+import fetch from 'node-fetch';
+import { keys } from './sources/keys.js';
+
+const app = express();
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send('hello from backend to frontend!');
+});
+
+app.post('/weather/:cityName', async (req, res) => {
+  try {
+    const parCityName = req.params.cityName;
+    const apiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${parCityName}&units=metric&appid=${keys.API_KEY}`;
+
+    const fetchResponse = await fetch(apiUrl);
+    const json = await fetchResponse.json();
+    if (parseInt(json.cod) === 404) {
+      const errorText = { weatherText: 'City is not found!' };
+      res.json(errorText);
+    } else {
+      const weatherContent = {
+        cityName: parCityName,
+        Temperature: json.main.temp,
+      };
+      res.json(weatherContent);
+    }
+  } catch (error) {
+    console.error(error);
```

You should also return an error to the user! In this case they are sent nothing and aren't sure what to do next